flow360/component/simulation/framework/entity_registry.py

```diff
@@ -105,13 +107,21 @@ def find_by_naming_pattern(self, pattern: str) -> list[EntityBase]:
         # pylint: disable=no-member
         for entity_list in self.internal_registry.values():
             matched_entities.extend(filter(lambda x: regex.match(x.name), entity_list))
+
+        if not matched_entities and error_when_no_match is True:
+            raise ValueError(
+                f"No entity found in registry with given name/naming pattern: '{pattern}'."
+            )
+        if enforce_output_as_list is False and len(matched_entities) == 1:
+            return matched_entities[0]
+
         return matched_entities
 
-    def find_by_name(self, name: str):
+    def find_single_entity_by_name(self, name: str):
         """Retrieve the entity with the given name from the registry."""
-        entities = self.find_by_naming_pattern(name)
-        if not entities:
-            raise ValueError(f"No entity found in registry with given name: '{name}'.")
+        entities = self.find_by_naming_pattern(
+            name, enforce_output_as_list=True, error_when_no_match=True
+        )
```

Comment:
I think `get_entity(name=)` or `find_by_name(name=)` would be better. Can we ensure no 2 entities with the same name lands in registry? them we don't need this error handling

Reply:
It will be hard to prevent the same naming because we may have Surface entity and a Box entity for example have the same name. This is allowed.